File: ToolBag/ParseTool/src/parsetool/workflow/models/statements/Declare.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package parsetool.workflow.models.statements;

import parsetool.models.common.TokenPair;
import parsetool.workflow.models.Generic_Type;
import parsetool.workflow.models.expressions.Expression;

/**
 *
 * @author cqy
 */
public class Declare extends TokenPair{
    private Generic_Type generic_type;
    private String name;
    private Expression default_value;

    /**
     * @return the name
     */
    public String getName() {
        return name;
    }

    /**
     * @param name the name to set
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * @return the default_value
     */
    public Expression getDefault_value() {
        return default_value;
    }

    /**
     * @param default_value the default_value to set
     */
    public void setDefault_value(Expression default_value) {
        this.default_value = default_value;
    }

    /**
     * @return the generic_type
     */
    public Generic_Type getGeneric_type() {
        return generic_type;
    }

    /**
     * @param generic_type the generic_type to set
     */
    public void setGeneric_type(Generic_Type generic_type) {
        this.generic_type = generic_type;
    }
}

File: CodeHelper/CodeHelper.Core/Parse/ParseResults/Workflows/Statements/Declare.cs
using System;
using System.Collections.Generic;
using CodeHelper.Core.Parse.ParseResults.Workflows.Expressions;
namespace CodeHelper.Core.Parse.ParseResults.Workflows.Statements
{
    public class Declare : TokenPair
    {
        public Generic_Type Generic_Type { get; set; }
        public String Name { get; set; }
        public Expression Default_value { get; set; }

        public VariableInfo Parse()
        {
            //var rslt = new FieldInfo();
            //rslt.Name = this.Name;
            //rslt.Type_name = this.Type_name;
            //return rslt;

            var rslt = new VariableInfo(this);
            rslt.Name = this.Name;
            rslt.GenericType = this.Generic_Type.Parse();
            if (this.Default_value != null)
            {
                rslt.Value = Default_value.Parse();
            }
            return rslt;
        }
    }
}